lib/conjur/api/ldapsync.rb

```diff
@@ -70,22 +65,52 @@ def ldap_sync_show_profile(profile, options = {})
     #
     # @param [Hash] profile a hash containing the LDAP sync configuration
     # @param [Hash] options reserved
-    def ldap_sync_update_profile(profile, options = {})
+    def ldap_sync_update_profile(profile_name, profile, options = {})
       options[:json_config] = profile.to_json
-      resp = RestClient::Resource.new(Conjur.configuration.appliance_url, credentials)['ldap-sync']['config'].put(options.to_json, :content_type => 'application/json')
+      resp = RestClient::Resource.new(Conjur.configuration.appliance_url, credentials)['ldap-sync']['config'][profile_name].put(options.to_json, :content_type => 'application/json')
       JSON.parse(resp.body)
     end
 
+    # @api private
+    # Search using an LDAP sync profile
+    #
+    # @param [String] profile name
+    # @param [Hash] options reserved
+    def ldap_sync_search(profile, options = {})
+      headers = credentials.dup.tap {|h|
+        h[:headers][:accept] = 'text/event-stream'
+      }
+
+      response = RestClient::Resource.new(Conjur.configuration.appliance_url, headers)['ldap-sync']['search'].post(options.merge(:config_name => profile))
+      JSON.parse(get_json("groups", response)).merge('events' => find_log_events(response))
+    end
+
     # @!endgroup
     
     private
-    def find_policy_event(response)
-      response.body.lines.find {|l| l =~ /^data: {"policy":/}.try(:[], 6..-1)
+    def get_json(ok_key, response)
```

Comment:
It might be semantics, but my initial reaction to `ok_key` was that it was related to a response code. It looks like it's acting like as a filter to target the lines to stream.  What do you think about using a variable name that's a little more reflective of the action, maybe something `event_filter`, `log_filter`, or `event_type`? It might make it a bit easier to understand when scanning code.